### ISSUE TITLE
Automatically remove old checkpoints and visualizations

### ISSUE THREAD

**Issue by RunDevelopment** [muslll/neosr#66]
I recently trained 2 models with neosr and my hard drive filled up pretty quickly with old checkpoints and visualizations that were of no interest to me. I *do* want to see the visualizations of recent iterations and compare them with older ones, but I don't need every saved visualization to do that.

So I would to request for neosr to add an option for `train.py` to automatically remove old checkpoints and their visualizations to take up less space. I think the option could be called `--auto-clean` or similar.

Since I'm talking about checkpoints *and* visualizations in this issue, `--auto-clean` would only work when we can map visualizations to checkpoints. Basically, we need `val.val_freq == logger.save_checkpoint_freq`. If there are no visualizations, then that works too of course.  If the option is enabled, then only the following checkpoints (and associated visualizations) should be *kept*:

- Keep the latest 10 checkpoints.
- Keep every 10th checkpoint.
- Keep every checkpoint that has been the best in at least one validation metric.

All other checkpoints (and their associated validations) should be removed.

What do you think?

**Comment by muslll** [comment]
It's an interesting idea. However one can do that manually in a few seconds... so I would say this is low-priority for now. But if anyone sends a PR I can definitely take a look and merge it :+1:

**Comment by RunDevelopment** [comment]
> However one can do that manually in a few seconds...

One cannot. I have a few dozen validation images I test with. I'm not going to go through 50 or so folders just to hopefully delete the same amount of files in each.

**Comment by muslll** [comment]
Why would you keep that many val images anyway? Doesn't make sense, sorry. Validation images are there just to take a look at how the model is training. After you train the model, just delete it... I've been training for many years now, in very large quantities, never had that problem.

**Comment by RunDevelopment** [comment]
> Why would you keep that many val images anyway?

Mostly to get a feel for what the model is and is not good at. E.g. the last model I trained as a JPEG model, so I had a set of validation images and compressed all with different JPEG options. That left me with `(val images) x (JPEG options I checked)` images. 

> If you want to contribute, feel free to send a PR

Maybe. My main issue right now is that neosr code isn't particularly inviting. Entirely untyped + linter warnings every 5 lines isn't a great combo...
